packages/firebase_ml_vision/ios/Classes/BarcodeDetector.m

```diff
@@ -1,4 +1,199 @@
 #import "FirebaseMlVisionPlugin.h"
 
 @implementation BarcodeDetector
+static FIRVisionBarcodeDetector *barcodeDetector;
+
++ (void)handleDetection:(FIRVisionImage *)image result:(FlutterResult)result {
+  if (barcodeDetector == nil) {
```

Comment:
Small nitpick:
I believe `if (barcodeDetector) {` should work.

Reply:
Did you mean `if (!barcodeDetector) {` ?